contracts/TokenFactory.sol

```diff
@@ -135,10 +154,20 @@ contract TokenFactory is ReentrancyGuard, Ownable {
     }
 
     function buy(address tokenAddress) external payable nonReentrant {
+        require(
```

Comment:
require current competition id to equal token's creation competition id (or "competing" competition id, if later we want to allow winning tokens to re-enroll the competition)